cluster/src/main/java/com/hartwig/pipeline/calling/sage/SageGermlineCaller.java

```diff
@@ -0,0 +1,92 @@
+import com.hartwig.pipeline.Arguments;
+import com.hartwig.pipeline.ResultsDirectory;
+import com.hartwig.pipeline.alignment.AlignmentPair;
+import com.hartwig.pipeline.datatypes.DataType;
+import com.hartwig.pipeline.datatypes.FileTypes;
+import com.hartwig.pipeline.execution.PipelineStatus;
+import com.hartwig.pipeline.execution.vm.BashCommand;
+import com.hartwig.pipeline.execution.vm.BashStartupScript;
+import com.hartwig.pipeline.execution.vm.VirtualMachineJobDefinition;
+import com.hartwig.pipeline.execution.vm.VmDirectories;
+import com.hartwig.pipeline.execution.vm.unix.UnzipToDirectoryCommand;
+import com.hartwig.pipeline.metadata.SomaticRunMetadata;
+import com.hartwig.pipeline.reruns.PersistedDataset;
+import com.hartwig.pipeline.resource.ResourceFiles;
+import com.hartwig.pipeline.stages.SubStageInputOutput;
+import com.hartwig.pipeline.storage.RuntimeBucket;
+
+public class SageGermlineCaller extends SageCaller {
+
+    public static final String NAMESPACE = "sage_germline";
+
+    private final ResourceFiles resourceFiles;
+
+    public SageGermlineCaller(final AlignmentPair alignmentPair, final ResourceFiles resourceFiles,
+            final PersistedDataset persistedDataset) {
+        super(alignmentPair, persistedDataset, DataType.SOMATIC_VARIANTS_SAGE);
+        this.resourceFiles = resourceFiles;
+    }
+
+    @Override
+    public String namespace() {
+        return NAMESPACE;
+    }
+
+    @Override
+    public List<BashCommand> commands(final SomaticRunMetadata metadata) {
+
+        List<BashCommand> commands = Lists.newArrayList();
+        commands.add(new UnzipToDirectoryCommand(VmDirectories.RESOURCES, resourceFiles.snpEffDb()));
+
+        String tumorBamPath = getTumorBamDownload().getLocalTargetPath();
+        String referenceBamPath = getReferenceBamDownload().getLocalTargetPath();
+        String tumorSampleName = metadata.tumor().sampleName();
+        String referenceSampleName = metadata.reference().sampleName();
+
+        final SageCommandBuilder sageCommandBuilder =
+                new SageCommandBuilder(resourceFiles).germlineMode(referenceSampleName, referenceBamPath, tumorSampleName, tumorBamPath);
+        SageApplication sageApplication = new SageApplication(sageCommandBuilder);
+        SageGermlinePostProcess sagePostProcess = new SageGermlinePostProcess(referenceSampleName, tumorSampleName, resourceFiles);
+
+        SubStageInputOutput sageOutput = sageApplication.andThen(sagePostProcess).apply(SubStageInputOutput.empty(tumorSampleName));
+        commands.addAll(sageOutput.bash());
+
+        return commands;
+    }
+
+    @Override
+    public VirtualMachineJobDefinition vmDefinition(final BashStartupScript bash, final ResultsDirectory resultsDirectory) {
+        return VirtualMachineJobDefinition.sageGermlineCalling(bash, resultsDirectory);
+    }
+
+    @Override
+    public boolean shouldRun(final Arguments arguments) {
+        return arguments.runGermlineCaller();
```

Comment:
Can we add a new toggle here so we can turn it off without disabling the GATK germline calling?

Reply:
Done. Added run_sage_germline_caller argument.